packages/mds-audit/attachments.ts

```diff
@@ -0,0 +1,177 @@
+      accessKeyId: aws.config.credentials.accessKeyId,
+      region: s3Region
+    })
+  }
+})
+
+export const multipartFormUpload = multer({ storage: memoryStorage }).single('file')
+
+export function attachmentSummary(attachment: Attachment): AttachmentSummary {
+  const thumbnailUrl = attachment.thumbnail_filename ? attachment.base_url + attachment.thumbnail_filename : ''
+  return {
+    attachment_id: attachment.attachment_id,
+    attachment_url: attachment.base_url + attachment.attachment_filename,
+    thumbnail_url: thumbnailUrl
+  }
+}
+
+export function validateFile(file: Express.Multer.File) {
+  if (!file || file.buffer.byteLength === 0) {
+    throw new ValidationError('No attachment found')
+  } else if (path.extname(file.originalname).replace('.', '') === '') {
+    throw new ValidationError(`Missing file extension in filename ${file.originalname}`)
+  } else if (!supportedMimetypes.includes(file.mimetype)) {
+    throw new UnsupportedTypeError(`Unsupported mime type ${file.mimetype}`)
+  }
+}
+
+async function writeAttachmentS3(file: Express.Multer.File) {
+  // Process attachment and thumbnail and output to Buffers
+  const [attachmentBuf, thumbnailBuf] = await Promise.all([
+    sharp(file.buffer)
+      .rotate() // automatically based on EXIF
+      .toBuffer(),
+    sharp(file.buffer)
+      .rotate()
+      .resize(thumbnailSize, thumbnailSize, {
+        fit: sharp.fit.inside,
+        withoutEnlargement: true
+      })
+      .toBuffer()
+  ])
+
+  // Upload attachment and thumbnail to S3
+  const attachmentId = uuid()
+  const ext = path.extname(file.originalname).toLowerCase()
+  const attachmentFilename = attachmentId + ext
+  const thumbnailFilename = `${attachmentId}.thumbnail${ext}`
+  await Promise.all([
+    s3
+      .upload({
+        Bucket: s3Bucket,
+        Key: [s3BucketSubdir, attachmentFilename].join('/'),
+        Body: attachmentBuf,
+        ACL: s3ACL
+      })
+      .promise(),
+    s3
+      .upload({
+        Bucket: s3Bucket,
+        Key: [s3BucketSubdir, thumbnailFilename].join('/'),
+        Body: thumbnailBuf,
+        ACL: s3ACL
+      })
+      .promise()
+  ])
+  return {
+    attachment_filename: attachmentFilename,
+    attachment_id: attachmentId,
+    base_url: `https://${s3Bucket}.s3-${s3Region}.amazonaws.com/${s3BucketSubdir}/`,
+    mimetype: file.mimetype,
+    thumbnail_filename: thumbnailFilename,
+    thumbnail_mimetype: file.mimetype
+  } as Attachment
+}
+
+export async function writeAttachment(file: Express.Multer.File, auditTripId: UUID) {
+  try {
+    const attachment: Attachment = await writeAttachmentS3(file)
+    await db.writeAttachment(attachment)
+    await db.writeAuditAttachment({
+      attachment_id: attachment.attachment_id,
+      audit_trip_id: auditTripId
+    } as AuditAttachment)
+    return attachment
+  } catch (err) {
+    await log.error('writeAttachment error', err.stack || err)
+    return null
+  }
```

Comment:
To @avatarneil point ... might make sense to let this exception be handled by the caller.

Reply:
Agreed